uwblueprint/headsupguys
# feat: improve frontend auth connections across end user exp

Please review and merge #102 first
## Implementation description
* The tool layout (the banner with the logo) has a functional Sign Up / Out button which dynamically changes based on auth state
* Implemented the sign up / log in modal
* Unauthenticated users attempting to exit a module or certificate page will be greeted with the sign up or log in modal
https://www.loom.com/share/5518217103664d5ab6df0d302bb1052e

## Steps to test
1. Visit http://localhost:3000 as an authenticated or unauthenticated user
2. Attempt to log out and sign up (whichever order makes sense) and ultimately log out again
3. Go to a tool with a module and on various pages in the module attempt to exit the module via in platform clicks -> check whether you get the sign up or log in modal to block you from doing so
4. Check whether the modal acts in an intuitive way
5. Complete the module as an unauthenticated user and do Step 3 on the certificates page
6. Perform the above as an authenticated user and ensure the modal never pops up



## What should reviewers focus on?
* 


## Checklist
